src/main/java/uk/gov/hmcts/reform/et/syaapi/service/ManageCaseRoleService.java

```diff
@@ -0,0 +1,394 @@
+                 findCaseForRoleModificationRequest.getCaseSubmissionReference());
+        return null;
+    }
+
+    /**
+     * Modifies user case roles by the given modification type. Gets case assignment user roles request which has
+     * a list of case_users that contains case id, user id and case role to modify the case. For assigning a new role
+     * to the case modification type should be Assignment, to revoke an existing role modification type should be
+     * Revoke
+     * @param authorisation Authorisation token of the user
+     * @param modifyCaseUserRolesRequest This is the list of case roles that should be Revoked or Assigned to
+     *                                   respondent. It also has the idam id of the respondent and the case id
+     *                                   of the case that will be assigned
+     * @param modificationType this value could be Assignment or Revoke.
+     * @throws IOException Exception when any problem occurs while calling case assignment api (/case-users)
+     */
+    public void modifyUserCaseRoles(String authorisation,
+                                    ModifyCaseUserRolesRequest modifyCaseUserRolesRequest,
+                                    String modificationType)
+        throws IOException {
+        // Gets httpMethod by modification type. If modification type is Assignment, method is POST, if Revoke, method
+        // is DELETE. Null if modification type is different then Assignment and Revoke.
+        HttpMethod httpMethod = RemoteServiceUtil.getHttpMethodByCaseUserRoleModificationType(modificationType);
+        // Checks modifyCaseUserRolesRequest parameter if it is empty or not and it's objects.
+        // If there is any problem throws ManageCaseRoleException.
+        ManageCaseRoleServiceUtil.checkModifyCaseUserRolesRequest(modifyCaseUserRolesRequest);
+        if (MODIFICATION_TYPE_REVOKE.equals(modificationType)) {
+            setAllRespondentsIdamIdAndDefaultLinkStatuses(authorisation, modifyCaseUserRolesRequest, modificationType);
+        }
+        CaseAssignmentUserRolesRequest caseAssignmentUserRolesRequest =
+            ManageCaseRoleServiceUtil.generateCaseAssignmentUserRolesRequestByModifyCaseUserRolesRequest(
+                modifyCaseUserRolesRequest);
+        log.info(getModifyUserCaseRolesLog(caseAssignmentUserRolesRequest, modificationType, true));
+        ResponseEntity<CaseAssignmentUserRolesResponse> response;
+        try {
+            String adminToken = adminUserService.getAdminUserToken();
+            HttpEntity<CaseAssignmentUserRolesRequest> requestEntity =
+                new HttpEntity<>(caseAssignmentUserRolesRequest,
+                                 RemoteServiceUtil.buildHeaders(adminToken, this.authTokenGenerator.generate()));
+            log.info("********{}************", ccdApiUrl);
+            response = restTemplate.exchange(ccdApiUrl + ManageCaseRoleConstants.CASE_USERS_API_URL,
```

Comment:
I think this should be AAC as et-sya-api has been added to the whitelist for AAC and not CCD
```suggestion
            response = restTemplate.exchange(aacUrl + ManageCaseRoleConstants.CASE_USERS_API_URL,
```

Reply:
It doesn't matter Harpreet, I updated it yesterday because I was getting that error. But Updated it back to aacUrl.